### PR TITLE
[WIP] CentOS support

### PR DESCRIPTION
**Pull request by errordeveloper**
Still need to add docs and tidy up a few things (suggestions are welcome)!

### CODE REVIEW

File: ansible/roles/k8s-master/tasks/main.yml [hunk 1]
```diff
@@ -28,14 +29,37 @@
...
+    state: latest
+  with_items:
+    - jq
+  tags:
+    - k8s
+
 - name: Set --advertise-address flag in kube-apiserver static pod manifest (workaround for https://github.com/kubernetes/kubernetes/issues/34101)
   become: yes
-  shell: "jq '.spec.containers[0].command |= .+ [\"--advertise-address=172.42.42.1\"]' /etc/kubernetes/manifests/kube-apiserver.json > /tmp/kube-apiserver.json && mv /tmp/kube-apiserver.json /etc/kubernetes/manifests/kube-apiserver.json"
+  shell: "jq '.spec.containers[0].command |= .+ [\"--advertise-address=172.42.42.11\"]' /etc/kubernetes/manifests/kube-apiserver.json > /tmp/kube-apiserver.json && mv /tmp/kube-apiserver.json /etc/kubernetes/manifests/kube-apiserver.json"
```
Comment (errordeveloper) [inline ansible/roles/k8s-master/tasks/main.yml]: I should actually remove this, it's been fixed upstream.

**Comment by davidkbainbridge** [comment]
Does this change the default to centos as opposed to an ubuntu deployment?

File: ansible/roles/k8s-base/tasks/main.yml [hunk 2]
```diff
@@ -41,28 +44,31 @@
...
   tags:
     - k8s
 
-- name: Ensure Kubernetes Repository
+- name: Ensure Kubernetes APT Repository
+  when: ansible_distribution == 'Ubuntu'
   become: yes
   apt_repository:
-    repo: 'deb http://apt.kubernetes.io/ kubernetes-xenial main'
+    repo: 'deb http://apt.kubernetes.io/ kubernetes-xenial-unstable main'
```
Comment (davidkbainbridge) [inline ansible/roles/k8s-base/tasks/main.yml]: Do we really want to use an 'unstable' repo?

Comment (errordeveloper) [inline ansible/roles/k8s-base/tasks/main.yml]: Well spotted, this was a temporary work around!

File: Vagrantfile [hunk 3]
```diff
@@ -1,20 +1,28 @@
 # -*- mode: ruby -*-
 # # vi: set ft=ruby :
 
+boxes = {
+  ubuntu: "ubuntu/xenial64",
+  centos: "centos/7",
+}
+
+distro = :centos # :ubuntu
```
Comment (errordeveloper) [inline Vagrantfile]: Yes, happy to change the default.

File: ansible/roles/k8s-master/tasks/main.yml (hunk 1 above)
Comment (errordeveloper) [inline ansible/roles/k8s-master/tasks/main.yml]: I'm quite sure it is upstream, will test.

**Comment by errordeveloper** [comment]
Just getting back to this, people seem to be staring my fork as it's linked from the docs, let's fix it! I'll make the changes you've suggested above and test... do you think this still is broadly applicable?

**Comment by Chabane** [comment]
@errordeveloper, @davidkbainbridge  https://github.com/NirbyApp/generator-mitosis will support CentOS, Ubuntu and CoreOS 
If you are interested to contribute, do not hesitate :+1:

**Pull request closed without merging**